File: apps/nekbone/refactored/unit_tests/jacobf/nekbone_jacobf.c
#ifndef UNIT_TEST_NEKBONE_JACOBF_C
#define UNIT_TEST_NEKBONE_JACOBF_C

#include <stdio.h>
#include "polybasis.h"

#define Err_t int
#define IFEB if(err) break

int main(int argc, char * argv[])
{
    FILE * fout = 0;

    Err_t err=0;
    while(!err){
        fout = fopen("./ccode.out", "w");
        if(!fout) {err=__LINE__; IFEB;}

        double POLY,PDER, POLYM1,PDERM1, POLYM2,PDERM2;

        double alpha, beta;
        double x,sz;

        unsigned int podegree, firstdegree, lastdegree;
        unsigned int sampleCount,s;

        alpha = 0; // This is how it is use throughout NEKbone.
        beta  = 0; // This is how it is use throughout NEKbone.

        // If any of these values changes, make also the change in nekbone_jacobf.f90
        firstdegree = 2;
        lastdegree  = 16;
        sampleCount = 10;

        sz = sampleCount;
        for(podegree=firstdegree; podegree<=lastdegree; ++podegree){
            for(s=1; s<=sampleCount; ++s){
                x = 1;
                x = x /(sz +1);

                err = nbb_JACOBF(podegree, alpha,beta, x,
                        &POLY, &PDER, &POLYM1,&PDERM1, &POLYM2,&PDERM2); IFEB;

                fprintf(fout, "%10u%10u%23.14E%23.14E%23.14E%23.14E%23.14E%23.14E\n",
                        podegree,s,  POLY,PDER, POLYM1,PDERM1, POLYM2,PDERM2
                       );
            }
            IFEB;
        }
        IFEB;

        if(fout){
            fclose(fout); fout=0;
        }

        break; //while(!err){
    }

    if(fout){
        fclose(fout); fout=0;
    }

    if(err){
        printf("ERROR: %lu\n", (unsigned long)err );
        return 1;
    }
    return 0;
}

#endif // UNIT_TEST_NEKBONE_JACOBF_C

File: apps/nekbone/refactored/unit_tests/jacobf/nekbone_jacobf.f90
!-----------------------------------------------------------------------
! The code for the procedure "jacobf" was taken from
!   nekbone-2.3.4/src/speclib.f
!-----------------------------------------------------------------------
      program test_jacobf

      implicit none

      REAL*8    POLY,PDER, POLYM1,PDERM1, POLYM2,PDERM2

      REAL*8 alpha, beta
      REAL*8 x,sz

      integer podegree
      integer firstdegree, lastdegree
      integer sampleCount,s

      alpha = 0. ! This is how it is use throughout NEKbone
      beta  = 0. ! This is how it is use throughout NEKbone

      ! If any of these values changes, make also the change in nekbone_jacobf.c
      firstdegree = 2
      lastdegree  = 16
      sampleCount = 10

      open(unit=10,file='fortran.out')

      sz = ((sampleCount))
      do podegree=firstdegree,lastdegree,1
        do s = 1,sampleCount,1
            x = ((1))
            x = x /(sz + 1.)
            call JACOBF(POLY,PDER, POLYM1,PDERM1,POLYM2,PDERM2, podegree, alpha,beta,x)

            write(10,'(2I10,6ES23.14)') podegree, s, POLY,PDER, POLYM1,PDERM1, POLYM2,PDERM2
        enddo
      enddo

      close(10)

      call exit(0)
      end
!--------------------------------------------------------------------
!
!     Computes the Jacobi polynomial (POLY) and its derivative (PDER)
!     of degree N at X.
!
!       ALP must be > -1, is 0 in ZWGL
!       BET must be > -1, is 0 in ZWGL
!
!     -1 < X < 1
!--------------------------------------------------------------------
      SUBROUTINE JACOBF (oPOLY,oPDER,oPOLYM1,oPDERM1,oPOLYM2,oPDERM2,N,ALP,BET,X)

      IMPLICIT REAL*8  (A-H,O-Z)
      APB  = ALP+BET
      oPOLY = 1.
      oPDER = 0.
      IF (N .EQ. 0) RETURN
      POLYL = oPOLY
      PDERL = oPDER
      oPOLY  = (ALP-BET+(APB+2.)*X)/2.
      oPDER  = (APB+2.)/2.
      IF (N .EQ. 1) RETURN
      DO 20 K=2,N
         DK = ((K))
         A1 = 2.*DK*(DK+APB)*(2.*DK+APB-2.)
         A2 = (2.*DK+APB-1.)*(ALP**2-BET**2)
         B3 = (2.*DK+APB-2.)
         A3 = B3*(B3+1.)*(B3+2.)
         A4 = 2.*(DK+ALP-1.)*(DK+BET-1.)*(2.*DK+APB)
         POLYN  = ((A2+A3*X)*oPOLY-A4*POLYL)/A1
         PDERN  = ((A2+A3*X)*oPDER-A4*PDERL+A3*oPOLY)/A1
         PSAVE  = POLYL
         PDSAVE = PDERL
         POLYL  = oPOLY
         oPOLY   = POLYN
         PDERL  = oPDER
         oPDER   = PDERN
 20   CONTINUE
      oPOLYM1 = POLYL
      oPDERM1 = PDERL
      oPOLYM2 = PSAVE
      oPDERM2 = PDSAVE
      RETURN
      END
